js/src/components/tree-select-control/index.js

```diff
@@ -163,52 +271,107 @@ const TreeSelectControl = ( {
 
 		let filteredTreeOptions = cloneDeep( treeOptions );
 
-		if ( filter ) {
-			filteredTreeOptions = filteredTreeOptions.filter( filterOption );
-		}
-
```

Comment:
❔ 
Remove this checking will make the HTML structure of all `label` become the following when not searching. Perhaps making up the removed checking to another place would be nicer.

```html
<label class="components-checkbox-control__label">
  <span>
    <span></span>
    <strong></strong>
    <span>Spain</span>
  </span>
</label>
```